OneCommunityGlobal/HGNRest
# Xiao F Create way for owner to edit & update ChatGPT prompt text

# Description
![CleanShot 2023-11-10 at 11 19 59](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/43768723/fea012f3-77fd-4a53-9902-f52daa2bee9b)


## Related PRS (if any):
This back-end PR is related to the [#1510](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1510) front-end PR.

## Main changes explained:
- Create file `src/models/dashBoardData.js` for prompt text schema
- Update multiple files in controller and router for get and update prompt

## How to test:
1. Check into current branch
2. Make sure do `npm run build` and `npm start` to start the BE
3. Made sure logged in as Owner user
4. Copy Auth token from Right click -> Inspect -> Application -> Local storage -> http://localhost:3000 -> token
5. Go to Postman -> Select 'GET' http://localhost:4500/api/dashboard/aiPrompt -> Headers -> put Key 'Authorization', Value 'the token you copied in step 4'
6. Send GET request and check "aIPromptText": "xxxx" is returned
7. Change GET to PUT
8. Go to 'Body' -> Select 'raw' -> Type in `{
    "aIPromptText": "Updated prompt"
}` -> Send PUT request -> Check if 'Successfully saved AI prompt.' is returned and use GET to check if prompt is changed

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HGNRest/assets/43768723/8b0c6383-adb2-4b19-a378-5d8c99d37d2b


## Note:
N/A
